### ISSUE TITLE
os.geteuid and os.getegid are called unnecessarily in SSHPublicKeyDatabase.checkKey

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6405]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#6405|
|Type|defect|
|Created|2013-04-05 00:36:49Z|
|Branch|https://github.com/twisted/twisted/tree/effectively-pointless-6405|

These functions are called and their results saved, but nothing is ever done with them.

<details><summary>Searchable metadata</summary>

```
trac-id__6405 6405
type__defect defect
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_effectively_pointless_6405 branches/effectively-pointless-6405
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__conch conch
keywords__ 
time__1365122209000000 1365122209000000
changetime__1365199367000000 1365199367000000
version__None None
owner__exarkun exarkun
cc__z3p
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [37926]) Branching to 'effectively-pointless-6405'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun removed owner|
|-|-|

Just some deletions.  Unfortunately some tests were affected.  [Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/effectively-pointless-6405).

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_4"></a>|@tomprince set owner to @exarkun|
|-|-|

This looks good. Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set status to `closed`|
|-|-|

(In [37949]) Merge effectively-pointless-6405

Author: exarkun
Reviewer: tom.prince
Fixes: [#6405](https://github.com/twisted/twisted/issues/6405)

Remove some unnecessary code from the implementation of
`SSHPublicKeyDatabase.checkKey`.
